### ISSUE TITLE
Content warning system

### ISSUE THREAD

**Issue by XenonFiber** [Chocobozzz/PeerTube#1158]
Giving the user the choice to add warning text to videos would be good for things that don't need to be treated as NSFW, but still need a note of caution. This could also automatically be used when they federate to non-PeerTube instances with CW support.

Example design:
![image](https://user-images.githubusercontent.com/28912578/46312100-b86ad280-c589-11e8-8f43-d4e3410856fb.png)

The ability to hide the thumbnail could also be an option, as well as being able to automatically have them visible by default.

Example design:
![image](https://user-images.githubusercontent.com/28912578/46312363-7a21e300-c58a-11e8-81fa-2b562be3eb81.png)

**Comment by rigelk** [comment]
NSFW might seem limiting to graphic content, but (technically) it encompasses a wide array of topics: https://en.wikipedia.org/wiki/Not_safe_for_work ; I agree with you that its connotation is not ideal, though.

Implementation-wise, I'd see things this way: we could imagine renaming NSFW to CW, and add a topic line. If the topic line is left empty, we show the traditional "NSFW". We keep the default behaviour of blurring NSFW, and add a few *user*/*admin* preferences:
- hiding completely the thumbnail with black instead of just blurring
- blacklist/whitelist of keywords in a topic line. If the topic line matches a keyword of the blacklist for instance, it is treated and shown as "Sensitive Content" and hidden as previously defined.

**Comment by rigelk** [comment]
It seems related to the broader #784, too.

**Comment by XenonFiber** [comment]
@rigelk These are all really good ideas, but this is less about filtering and more about warning. It'd be used in a similar way to how they're used on Mastodon (which would also include the ability to add those CWs to a filter, if needed).

Unless I misunderstood, in that case, sorry.

**Comment by FirePowi** [comment]
Just to mention that I searched for a "content warning" enhancement before sending one if there wasn't.
And I guess this is really something we might need.

For now, "NSFW" is being used as a "Warning" but without the topic mentioned. That's like saying "This video contain some stuff you might not want to see… Maybe porn, maybe guns, maybe violence, maybe…".

I suggest something that look like that : 
![image](https://user-images.githubusercontent.com/14030913/58467594-4e1c2900-813c-11e9-99ee-f96ba7b08141.png)

Added to that :
![image](https://user-images.githubusercontent.com/14030913/58467657-5ffdcc00-813c-11e9-8d16-ac8bbd388d5a.png)
